src/cmd/api/Authentication.py

```diff
@@ -0,0 +1,35 @@
+import bcrypt
+import csv
+
+
+class authentication:
+    def register(self, username, password):
+
+        with open("../../core/storage/users.csv", 'r') as readFile:
+            reader = csv.reader(readFile, delimiter=' ')
+            for row in reader:
+                try:
+                    print(row[0].split(",")[0])
+                    if row[0].split(",")[0] == username:
+                        print("Username already exists")
+                        return False
+                except:
+                    pass
+
+            with open("../../core/storage/users.csv", 'a', newline='') as writeFile:
+                writer = csv.writer(writeFile)
+                salt = bcrypt.gensalt()
+                hashed_password = bcrypt.hashpw(password.encode('utf-8'), salt)
+                writer.writerow([username, hashed_password])
+
+        return True
+
+    def Login(self, username, password):
+        with open("../../core/storage/users.csv", 'r') as readFile:
+            reader = csv.reader(readFile)
+            for row in reader:
+                if row[0] == username:
+                    password_bytes = password.encode('utf-8')
+                    hashed_password = row[1].encode('utf-8')[2:-1]
+                    check = bcrypt.checkpw(password_bytes, hashed_password)
+                    return check
```

Comment:
we're missing a method to check if the token is valid on request. let's say I want to access some data from the route `/api/revenues`. We don't really want outsiders to be able to access that. So, when we make a request we check if the token is correct and is from this server and then we can process the request. something like `def ValidateToken` would do the trick

Reply:
je kan @token_required boven een function zetten. dit checked of de user een valid jwt token heeft